packages/framework/src/Foundation/Concerns/ManagesHydeKernel.php

```diff
@@ -71,6 +72,13 @@ public function getMediaOutputDirectory(): string
         return ltrim($this->getMediaDirectory(), '_');
     }
 
+    public function navigation(): NavigationMenu
+    {
+        $this->needsToBeBooted();
```

Comment:
This may be undesirable. It means that if someone tries to add to the menu from their code before the kernel is booted this could cause premature booting. Instead we may want to throw a helpful exception.